### DISCUSSION TITLE
Make Storage Adapters an extension type

### DISCUSSION THREAD

**Discussion by benhaynes** [directus/directus#3947] [Archive]
Check before 9.0.0

**Comment by Attacler** [comment]
@benhaynes is there any news on this part?

**Reply by benhaynes** [reply]
No, not yet. Support for S3 (and compatible) as well as Local Filesystem has proven to be what most people need. That said, abstracting this away will be great... and we'll want to add more App features at the same time (choosing storage adapters on upload, better File Library organization to embrace this, etc).

**Comment by rijkvanzanten** [comment]
Lets close this one in favor of https://github.com/directus/directus/discussions/14742, as that has a way more in depth discussion on the same topic 👍🏻
